### PR TITLE
fix: prevents putting meaningless tasks into the queue

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-270

### CODE REVIEW

File: model/Index/Service/ResultIndexer.php [hunk 1]
```diff
@@ -41,6 +42,9 @@ public function setNormalizer(NormalizerInterface $normalizer): self
 
     public function addIndex($resource): void
     {
+        if (!$this->getServiceLocator()->get(AdvancedSearchChecker::class)->isEnabled()) {
+            return;
+        }
```
Comment (gabrielfs7) [inline model/Index/Service/ResultIndexer.php:47]:
```suggestion
        }
        
```

File: tests/Unit/Index/Service/ResultIndexerTest.php [hunk 2]
```diff
@@ -41,18 +42,22 @@ class ResultIndexerTest extends TestCase
 
     /** @var QueueDispatcherInterface|MockObject */
     private $queueDispatcher;
+    /** @var AdvancedSearchChecker|MockObject  */
```
Comment (gabrielfs7) [inline tests/Unit/Index/Service/ResultIndexerTest.php:45]: PSR + our code style agreement

```suggestion

    /** @var AdvancedSearchChecker|MockObject  */
```

**Review by gabrielfs7 — CHANGES_REQUESTED** [review]
What about the other indexers in this repository?

**Comment by boajer** [comment]
> What about the other indexers in this repository?

But `\oat\taoAdvancedSearch\model\Index\Service\ResultIndexer` so far the only existing implementation of `\oat\taoAdvancedSearch\model\Index\Service\IndexerInterface`

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**